lib/api/src/js/js_import_object.rs

```diff
@@ -0,0 +1,71 @@
+use crate::js::{Export, ExternType, Module, NamedResolver};
+use std::collections::HashMap;
+
+/// This struct is used in case you want to create an `Instance`
+/// of a `Module` with imports that are provided directly from
+/// Javascript with a JS Object.
+#[derive(Clone, Default)]
+pub struct JsImportObject {
+    module_imports: HashMap<(String, String), ExternType>,
+    object: js_sys::Object,
+}
+
+unsafe impl Send for JsImportObject {}
+unsafe impl Sync for JsImportObject {}
+
+impl JsImportObject {
+    /// Create a new `JsImportObject`.
```

Comment:
There is no explanation of what the `module` parameter does. From the implementation I am guess that it is used to assign a type for each import, but this is not obvious from the documentation.